src/wallet/wallet.cpp

```diff
@@ -3032,6 +3091,20 @@ bool CWallet::CommitTransaction(CTransactionRef tx, mapValue_t mapValue, std::ve
             // Take key pair from key pool so it won't be used again
             reservekey.KeepKey();
 
+            // We determine if used before adding to wallet, as that WILL mark
+            // the inputs as used; after adding to wallet, we mark new UTXO's
+            // as used as appropriate
+            bool used = false;
+            if (IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE)) {
+                for (const auto& vin : wtxNew.tx->vin) {
+                    if (IsUsedDestination(vin.prevout.hash, vin.prevout.n)) {
+                        used = true;
```

Comment:
I don't understand what this code block is doing. You have a boolean `used` which is initially set to false and then may be updated to true, but you never use the value of `used` anywhere.

Reply:
That was leftovers from the incorrect code that I removed earlier. Thanks, removed!